packages/flutter_tools/templates/ffi_plugin/lib/projectName.dart.tmpl

```diff
@@ -0,0 +1,70 @@
+{{#no_platforms}}
+// You have generated a new plugin project without
+// specifying the `--platforms` flag. A plugin project supports no platforms is generated.
+// To add platforms, run `flutter create -t plugin_ffi --platforms <platforms> .` under the same
+// directory. You can also find a detailed instruction on how to add platforms in the `pubspec.yaml` at https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms.
+{{/no_platforms}}
+
+import 'dart:async';
+import 'dart:ffi';
+import 'dart:io';
+import 'dart:isolate';
+
+const libName = '{{projectName}}';
+
+final DynamicLibrary _dylib = () {
+  if (Platform.isMacOS || Platform.isIOS) {
+    return DynamicLibrary.open('$libName.framework/$libName');
+  }
+  if (Platform.isAndroid || Platform.isLinux) {
+    return DynamicLibrary.open('lib$libName.so');
+  }
+  if (Platform.isWindows) {
+    return DynamicLibrary.open('$libName.dll');
+  }
+  throw UnsupportedError('Unimplemented!');
+}();
+
+final sum = _dylib.lookupFunction<IntPtr Function(IntPtr, IntPtr),
+    int Function(int, int)>('sum');
+
+bool _initializedDartApiDL = false;
+void _ensureInitializedDartApiDL() {
+  if (_initializedDartApiDL) {
+    return;
+  }
+  final initializeApi = _dylib.lookupFunction<IntPtr Function(Pointer<Void>),
+      int Function(Pointer<Void>)>("Dart_InitializeApiDL");
+  final initializeResult = initializeApi(NativeApi.initializeApiDLData);
+  if (initializeResult != 0) {
+    throw "Failed to initialize API.";
+  }
+  if (_dylib.lookup<IntPtr>("Dart_PostCObject_DL").value == 0) {
+    throw "Failed to initialize API.";
+  }
+}
+
+final _sumAsync = () {
+  _ensureInitializedDartApiDL();
+  return _dylib.lookupFunction<IntPtr Function(Int64 port, IntPtr, IntPtr),
+      int Function(int port, int, int)>('sum_async');
+}();
+
+Future<int> sumAsync(int a, int b) async {
+  final completer = Completer<int>();
+  // Note: if this an often called function, it would be better to reuse the
+  // native port rather than creating a new one on every call.
+  final port = ReceivePort()
+    ..listen((dynamic message) {
+      if (message is int) {
+        completer.complete(message);
+      } else {
+        completer.completeError('Did not get an integer as message: $message.');
+      }
+    });
+  _sumAsync(port.sendPort.nativePort, a, b);
+  return completer.future.then((value) {
+    port.close();
+    return value;
+  });
+}
```

Comment:
consider adding more comments to this file to explain what's going on in more details.